src/manage_data.jl

```diff
@@ -3,16 +3,16 @@
 function readSMAT(filename::AbstractString)
     (rows,header) = readdlm(filename;header=true)
     A = sparse(
-               convert(Array{Int64,1},rows[1:parse(Int,header[3]),1])+1, 
-               convert(Array{Int64,1},rows[1:parse(Int,header[3]),2])+1, 
+               convert(Array{Int64,1},rows[1:parse(Int,header[3]),1]).+1, 
```

Comment:
Just use Int64.() instead

Reply:
Also, make sure this doesn't actually allocate two arrays when we add one... let's not be totally inefficient :)

Reply:
Sure! I've revisited that function and made it as follows:
```
function readSMAT(filename::AbstractString)
    rows,header = readdlm(filename;header=true)
    A = sparse(
               Int64.(rows[1:parse(Int,header[3]),1]).+1, 
               Int64.(rows[1:parse(Int,header[3]),2]).+1, 
               Int64.(rows[1:parse(Int,header[3]),3]),
               parse(Int,header[1]), 
               parse(Int,header[2])
               )
    return A
end
``` 
Then benchmarked it against:
```
function readSMAT2(filename::AbstractString)
    f = open(filename)
    line1 = readline(f)
    line1vals = split(line1," ")
    edgecount = parse(Int,line1vals[3])
    rows = zeros(Int64,edgecount,3)
    @inbounds for i = 1:edgecount
        curline = readline(f)
        curlinevals = split(curline," ")
        rows[i,:] = parse.(Int,curlinevals)
    end
    close(f)
    rows[:,1:2] .+= 1 # now rows has the correct values
    A = sparse(
               rows[:,1], 
               rows[:,2], 
               rows[:,3],
               parse(Int,line1vals[1]), 
               parse(Int,line1vals[2])
               )
    return A
end
```
And the latter is much faster... This is one of the larger examples
```
julia> @btime A1 = MatrixNetworks.readSMAT("/Users/hudanassar/.julia/dev/MatrixNetworks/data/U3A.smat");
  355.528 ms (1797607 allocations: 46.43 MiB)

julia> @btime A2 = MatrixNetworks.readSMAT2("/Users/hudanassar/.julia/dev/MatrixNetworks/data/U3A.smat");
  32.615 ms (233282 allocations: 14.82 MiB)
```
The .+ is non-allocating here... I'm sticking with the second version
Also, I had thought that convert is more efficient but that would only be true if the array had Int values... Example:
```
julia> x = floor.(rand(10_000)*100);

julia> y = floor.(Int64,rand(10_000)*100);

julia> typeof(x)
Array{Float64,1}

julia> typeof(y)
Array{Int64,1}

julia> @btime convert(Vector{Int64},$x);
  18.324 μs (2 allocations: 78.20 KiB)

julia> @btime convert(Vector{Int64},$y);
  2.371 ns (0 allocations: 0 bytes)

julia> @btime Int64.($x);
  19.776 μs (2 allocations: 78.20 KiB)

julia> @btime Int64.($y);
  7.383 μs (2 allocations: 78.20 KiB)
```

Reply:
Since we are changing this, why not use the following?

~~~~
    f = open(filename)
    header = readline(f)
    parts = split(header," ")
    nedges = parse(Int,parts[3])
    ei = zeros(Int64,edgecount)
    ej = zeros(Int64, edgecount)
    ev = zeros(Float64, edgecount)
    @inbounds for i = 1:edgecount
        curline = readline(f)
        parts = split(curline," ")
        ei[i] = parse(Int, parts[1])+1
        ej[i] = parse(Int, parts[2])+1
        ev[i] = parse(Float64, parts[3])
    end
    close(f)
    A = sparse(ei, ej, ev
               parse(Int,header[1]), 
               parse(Int,header[2])
               )
~~~~

Reply:
Sure! Changed here e78c50bb1ca0413f5207c6f904ee066672d4ee99